modules/azure/client_factory_test.go

```diff
@@ -0,0 +1,106 @@
+// /*
+
+// This file contains unit tests for the client factory implementation(s).
+
+// */
+
+package azure
+
+import (
+	"os"
+	"testing"
+
+	autorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Local consts for this file only
+var GovCloudEnvName = "AzureUSGovernmentCloud"
+var PublicCloudEnvName = "AzurePublicCloud"
+var ChinaCloudEnvName = "AzureChinaCloud"
+var GermanyCloudEnvName = "AzureGermanCloud"
+
+func TestDefaultEnvIsPublicWhenNotSet(t *testing.T) {
+	// Get a client factory
+	factory := multiEnvClientFactory{}
+
+	// save any current env value and restore on exit
+	currentEnv := os.Getenv(AzureEnvironmentEnvName)
+	defer os.Setenv(AzureEnvironmentEnvName, currentEnv)
+
+	// Set env var to gov
+	os.Setenv(AzureEnvironmentEnvName, "")
```

Comment:
Comment says "gov", but you're setting it to an empty string?

Reply:
Bad comment - resolved